src/Core/Streams/ResourceStream.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace AsyncAws\Core\Streams;
+
+use AsyncAws\Core\Exception\InvalidArgument;
+
+/**
+ * Convert a resource into a Stream.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class ResourceStream implements Stream
+{
+    /**
+     * @var resource
+     */
+    private $content;
+
+    private $chunkSize;
+
+    public function __construct($content, int $chunkSize = 64 * 1024)
+    {
+        if (!\is_resource($content)) {
+            throw new InvalidArgument(sprintf('Expect $content to be a resource, "%s" given.', \is_object($content) ? \get_class($content) : \gettype($content)));
+        }
+        if (!stream_get_meta_data($content)['seekable']) {
+            throw new InvalidArgument(sprintf('The give body is not seekable.'));
+        }
+        $this->content = $content;
+        $this->chunkSize = $chunkSize;
+    }
+
+    public function length(): ?int
+    {
+        return fstat($this->content)['size'] ?? null;
+    }
+
+    public function stringify(): string
+    {
+        if (-1 === fseek($this->content, 0)) {
+            throw new InvalidArgument('Unable to seek the content.');
+        }
+
+        return \stream_get_contents($this->content);
```

Comment:
What if we got a "read once" stream? Should we care? Will it work?

Reply:
The stream must be seekable (checked in constructor). Can a stream be "read once" and "Seakable"?